qiling/os/posix/syscall/select.py

```diff
@@ -41,13 +41,18 @@ def set_fd_set(buf, idx):
     tmp_w_fd, tmp_w_map = parse_fd_set(ql, _newselect_nfds, _newselect_writefds)
     tmp_e_fd, tmp_e_map = parse_fd_set(ql, _newselect_nfds, _newselect_exceptfds)
 
+    n = ql.archbit // 8 # 4 for 32-bit , 8 for 64-bit
+
     if _newselect_timeout != 0:
-        timeout = ql.unpack32(ql.mem.read(_newselect_timeout, 4))
+        timeout_p = ql.unpack(ql.mem.read(_newselect_timeout, n))
```

Comment:
maybe you should rename timeout_p  to a more proper name too ?

Reply:
timeout_ptr? What do you suggest?I am quite open to names :)

Reply:
i did not check yet. is that a pointer ? if that is a pointer then ptr is ok.

Reply:
updated. yes, it is the pointer of struct timeval *.